pkg/virtctl/imageupload/imageupload.go

```diff
@@ -617,12 +618,20 @@ func ensurePVCSupportsUpload(client kubernetes.Interface, pvc *v1.PersistentVolu
 	return pvc, nil
 }
 
-func getAndValidateUploadPVC(client kubernetes.Interface, namespace, name string, shouldExist bool) (*v1.PersistentVolumeClaim, error) {
+func getAndValidateUploadPVC(client kubecli.KubevirtClient, namespace, name string, shouldExist bool) (*v1.PersistentVolumeClaim, error) {
 	pvc, err := client.CoreV1().PersistentVolumeClaims(namespace).Get(context.Background(), name, metav1.GetOptions{})
 	if err != nil {
+		fmt.Printf("PVC %s/%s not found \n", namespace, name)
 		return nil, err
 	}
 
+	if !createPVC {
+		_, err = client.CdiClient().CdiV1alpha1().DataVolumes(namespace).Get(context.Background(), name, metav1.GetOptions{})
+		if err != nil {
+			return nil, fmt.Errorf("No DataVolume is associated with the existing PVC %s/%s", namespace, name)
```

Comment:
Is the "not found" the only error that can happen here? If not it would make sense to transport the error message to the user here.

Reply:
@dhiller changed the handling of the error in case of dv not found we will return this nice error, if not just the returned error. please check